### PR TITLE
Update core to latest, fix new byte array representation for ids

### PR DESCRIPTION
**Pull request by bogdandrutu**
Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

**Review by tigrannajaryan — APPROVED** [review]
LGTM, approving, but don't merge immediately, give codeowners time to review.

Codeowners, please make sure to review changes in your components.

File: processor/groupbytraceprocessor/event_test.go
Comment (jpkrohling) [inline processor/groupbytraceprocessor/event_test.go:51]: In the batchpertrace, you did this:

```
pdata.NewTraceID([16]byte{1, 2, 3, 4})
```

Can we keep it like that here as well? IMO, it's easier to visualize the differences between the cases.

File: processor/groupbytraceprocessor/processor.go
Comment (jpkrohling) [inline processor/groupbytraceprocessor/processor.go:157]: I think `Bytes() != nil` is far more readable than `IsValid()`, at least for this case here. Perhaps `IsNil()` would be a better name, if all that `IsValid()` does is checking whether the byte array is nil ?

**Review by jpkrohling — COMMENTED** [review]
LGTM, except for the `Bytes() != nil`  -> `IsValid()`.

File: processor/groupbytraceprocessor/processor.go
Comment (bogdandrutu) [reply processor/groupbytraceprocessor/processor.go:157]: a byte array can never be nil, it can be empty (all zeros). so the check for "!= nil" does not make any sense.

Comment (jpkrohling) [reply processor/groupbytraceprocessor/processor.go:157]: `IsEmpty()` then?

File: processor/groupbytraceprocessor/event_test.go
Comment (bogdandrutu) [reply processor/groupbytraceprocessor/event_test.go:51]: Done.

File: processor/groupbytraceprocessor/processor.go
Comment (bogdandrutu) [reply processor/groupbytraceprocessor/processor.go:157]: That is something that needs to get to core, but will consider that name. Can you file an issue?

Comment (jpkrohling) [reply processor/groupbytraceprocessor/processor.go:157]: Sure: https://github.com/open-telemetry/opentelemetry-collector/issues/2014

Comment (bogdandrutu) [reply processor/groupbytraceprocessor/processor.go:157]: This should not block the merge, as pointed in the issue `traceID.IsValid()` is equivalent with `!traceID.IsEmpty()`

File: Makefile
Comment (jpkrohling) [inline Makefile:62]: Should this really be part of this PR?

Comment (bogdandrutu) [reply Makefile:62]: I used it to debug lint errors, I will split

Comment (bogdandrutu) [reply Makefile]: Done.

**Review by jpkrohling — APPROVED** [review]
I reviewed the modules that I own and it looks good.

**Pull request merged**
